python-checks-testkit/src/main/java/org/sonar/python/checks/utils/PythonQuickFixVerifier.java

```diff
@@ -0,0 +1,105 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.utils;
+
+import com.sonar.sslr.api.AstNode;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import org.sonar.plugins.python.api.IssueLocation;
+import org.sonar.plugins.python.api.PythonCheck;
+import org.sonar.plugins.python.api.PythonCheck.PreciseIssue;
+import org.sonar.plugins.python.api.PythonFile;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.PythonVisitorContext;
+import org.sonar.plugins.python.api.tree.FileInput;
+import org.sonar.python.SubscriptionVisitor;
+import org.sonar.python.parser.PythonParser;
+import org.sonar.python.quickfix.IssueWithQuickFix;
+import org.sonar.python.semantic.ProjectLevelSymbolTable;
+import org.sonar.python.tree.PythonTreeMaker;
+
+public class PythonQuickFixVerifier {
+  private PythonQuickFixVerifier() {
+  }
+
+  private static List<PreciseIssue> scanFileForIssues(PythonCheck check, PythonVisitorContext context) {
+    check.scanFile(context);
+    if (check instanceof PythonSubscriptionCheck) {
+      SubscriptionVisitor.analyze(Collections.singletonList((PythonSubscriptionCheck) check), context);
+    }
+    return context.getIssues();
+  }
+
+  public static List<PreciseIssue> getIssuesWithQuickFix(String codeWithIssue, PythonCheck check) {
+    PythonParser parser = PythonParser.create();
+    PythonQuickFixFile pythonFile = new PythonQuickFixFile(codeWithIssue);
+    AstNode astNode = parser.parse(pythonFile.content());
+    FileInput parse = new PythonTreeMaker().fileInput(astNode);
+
+    PythonVisitorContext visitorContext = new PythonVisitorContext(parse,
+      pythonFile, null, "",
+      ProjectLevelSymbolTable.empty());
+
+    return scanFileForIssues(check, visitorContext);
+  }
+
+  public static String applyQuickFix(String codeWithIssue, IssueWithQuickFix issueWithQuickFix) {
+    IssueLocation loc = issueWithQuickFix.getQuickFixes().get(0).getTextEdits().get(0).issueLocation;
+    String replacement = loc.message();
+    int start = convertPositionToIndex(codeWithIssue, loc.startLine(), loc.startLineOffset());
+    int end = convertPositionToIndex(codeWithIssue, loc.endLine(), loc.endLineOffset());
+    return codeWithIssue.substring(0, start + 1) + replacement + codeWithIssue.substring(end + 1);
```

Comment:
I would prefer the `+ 1` to be handled by `convertPositionToIndex` because I don't like it to return an index such as `-1`.